internal/monitors/conviva/accountservice.go

```diff
@@ -0,0 +1,223 @@
+package conviva
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+//const (
+//	metricLensURLFormat = "https://api.conviva.com/insights/2.4/metrics.json?metrics=%s&account=%s&filter_ids=%s&metriclens_dimension_id=%d"
+//)
+
+var anyErrors bool
+var isInitialized bool
+
+// Account for Conviva account data
+type Account struct {
+	id                   string
+	Name                 string
+	filters              map[string]string
+	metricLensFilters    map[string]string
+	metricLensDimensions map[string]float64
+}
+
+// AccountService interface for Account related methods
+type AccountService interface {
+	GetDefault()                                  *Account
+	GetMetricLensDimensionMap(accountName string) map[string]float64
+	GetID(accountName string)                     string
+	GetFilters(accountName string)                map[string]string
+	GetMetricLensFilters(accountName string)      map[string]string
+	GetFilterID(accountName string, filterName string) string
+	GetMetricLensDimensionID(accountName string, metricLensDimension string) float64
+}
+
+type accountServiceImpl struct {
+	defaultAccount *Account
+	accounts       []*Account
+	ctx            context.Context
+	timeout        *time.Duration
+	httpClient     *HTTPClient
+}
+
+// NewAccountService factory function creating AccountService
+func NewAccountService(ctx context.Context, timeout *time.Duration, httpClient *HTTPClient) AccountService {
+	service := accountServiceImpl{ctx: ctx, timeout: timeout, httpClient: httpClient,}
+	service.init()
```

Comment:
Do the init with all of the HTTP requests on the initial read callback in case Conviva is having an outage when somebody starts up the agent.  Don't assume this will always be a one time thing that will always succeed.